### ISSUE TITLE
Cinnamon CPU usage reaches 100% after lock screen

### ISSUE THREAD

**Issue by snr-lab** [linuxmint/cinnamon#11754]
### Distribution

Linux Mint 21.1 Cinnamon Desktop

### Package version

5.6.8

### Graphics hardware in use

None

### Frequency

Always

### Bug description

Recently I have converted my Laptop to a Proxmox server and I am running Linux Mint Cinnamon in a container. When the screen is unlocked and no specific application is running, the CPU usage is 2-5%. As soon as lock the screen it reaches 96-98% even though I am not running any application. Once I unlock the screen again it goes down to 2-5%.

This issue is replicable only in Linux Mint Cinnamon Desktop. Linux Mint XFCE Desktop works fine.

I have provided more details about the system in this forum [https://forums.linuxmint.com/viewtopic.php?p=2348954](url)

### Steps to reproduce

1. Install Proxmox on a desktop.
2. Install Linux Mint Cinnamon OS in a VM and update it.
3. Log in to the system and the lock the screen.

### Expected behavior

The CPU usage should be similar for both Locked and Unlocked screen.

### Additional information

[https://forums.linuxmint.com/viewtopic.php?p=2348954](url)

**Comment by mtwebster** [comment]
Hi,

Are you using any type of display scaling inside the vm? If so can you set it back to 100% temporarily and test this again?

**Comment by snr-lab** [comment]
Yes, display scaling was 125% inside the VM. Setting it to 100% improves CPU usage. Now the screen is locked CPU usage is 30-45% when no application is running. Unlocking the screen CPU usage is still 2-3%.

One more observation, when I am trying to change the scaling from 125% to 100% after pressing apply button the screen gets black. It recovers after some time but the Display application not responding. I was able to change the scale after multiple trials and errors.

It seems there are some issues in display scaling.

**Comment by snr-lab** [comment]
Any suggestion to fix it?

**Comment by mtwebster** [comment]
The 100% cpu with scaling is fixed in cinnamon-screensaver 5.8. As for it still being 30%, I'm not sure - I can't reproduce this in virtualbox, there it's 0-1% while active.

The display settings issue, I can't reproduce either, it works reliably.

You can run the screensaver with debugging output:
```
cinnamon-screensaver --debug --hold
```
The debug flag will print a lot of extra info, the hold flag keeps the screensaver running indefinitely (it normally exits after 30 seconds of non-use).

You can also try running it with the `--no-fallback` flag (this disables a separate backup window that takes over in the event the main process dies).

You can activate the screensaver normally, or from the command line - `cinnamon-screensaver-command --lock` or `--activate`

I can have a look at the output from that. I would also suggest trying out the new Mint 21.2 iso, mainly for the fact that any fixes or updates are going to be to Cinnamon's 5.8+ line, not 5.6.

**Comment by snr-lab** [comment]
Hi @mtwebster 
Thanks for your response.

I tried with Mint 21.2 (Cinnamon 5.8.4) and it has same problem. Please find attachment fot the output of "cinnamon-screensaver --debug --hold"
[debug.txt](https://github.com/linuxmint/cinnamon/files/12087294/debug.txt)

Please let me know if I can provide any more details.

**Comment by snr-lab** [comment]
Any update on this bug?

**Comment by mtwebster** [comment]
No I'm not sure yet what might be causing this.

I realize this may be tedious, but it might be worth trying Mint 20.3

Starting with Mint 21, there was a major rework of the window manager (muffin) and cinnamon. It has caused some ongoing issues with the screensaver (mainly when using fractional scaling) - these are mostly fixed now, but there could be some other behavior that was introduced, but less common.

If it works ok, it may be worth sticking with it for now, if the screensaver is a blocker for you using this system. Obviously this would result in an older package base overall, so it'd be a tradeoff.

**Comment by snr-lab** [comment]
I tried Mint 20.3 and it does not have a problem. But I don't want to use it as the packages are too old.

In my use case, I do not need the screen saver at all. I just want to lock the screen with CPU usage is normal. It is fine if the screen turns off or black once I lock the screen. It should give me a password option when I press enter or click.

If you can help me with that it will be great.

**Comment by snr-lab** [comment]
@mtwebster  any help?

**Comment by mtwebster** [comment]
For now, no, unfortunately. I planned to set Proxmox up in the next few days to hopefully try and reproduce this. If you're seeing it I'm sure others are as well though they may not be aware of it.

**Comment by mtwebster** [comment]
Ok I got it set up and made a mint 21.2 guest, it seems to run fine when the screensaver is active.  I'm not sure what the rest of your configuration was (and I've never used this hypervisor before), but this is working ok:
![image](https://github.com/linuxmint/cinnamon/assets/262776/765961f6-24de-4638-b243-6fc87763f1ad)
![image](https://github.com/linuxmint/cinnamon/assets/262776/10126363-df29-419d-8a10-74eb5d88ec37)
![image](https://github.com/linuxmint/cinnamon/assets/262776/25b46b40-26e6-47e8-9142-b062b09b563a)

I tried a couple of different graphic cards - default, vmware-compatible, VirtIO-GPU, and they all seem to behave the same.
I didn't try this with any scaling, just at 1920x1080 and under. Also pretty much the same with only 1 core configured.

**Comment by snr-lab** [comment]
Thanks @mtwebster for taking the effort. Yes, it is working fine in your system. In my case even the scaling is 100%, the CPU usage is 30-40%. Unfortunately, I got some issue in power supply in my hardware. I will share my configuration or any other useful information as soon as the hardware issue is resolved.

**Comment by dceldran** [comment]
I have the same problem, i have a mini-pc with linux mint 21.2 and cinnamon 5.8.4 installed. When the lock screen activates, cpu raises to 60-70%.

I don't have any monitor attached to the mini-pc, i only connect through xrdp.

The steps to reproduce (in my case) are: 

1º Connect throught rdp to linux mint
2º Enable lock screen

And cpu goes to 60-70%. If i reconnect with rdp and i put my password to disable the lock screen, cpu downs to normal.

The only thing i can say is i only can reproduce the problem if i don't have any apps opened and then lock the screen. If i have opened, for example, system settings and then i lock the screen, the cpu don't raise to 60-70% and stays down, but when i have all apps closed and i lock screen, problem appears.

My workaround is to disable lock screen in System settings -> Screen Saver

**Comment by snr-lab** [comment]
Hi @mtwebster

My older system faced motherboard failure. I have purchased a new system with decent hardware configuration. The new system have capability to passthrough the GPU.

<img width="689" alt="SystemResource" src="https://github.com/linuxmint/cinnamon/assets/19688360/336d34ff-6d65-4e07-a565-4dec4a70aa3f">

With the new hardware system performs much better. But the core issue remain same. When system is idle the cpu usage is 0.15% - 0.22%. But as soon as the screen saver is appears the cpu usage reaches to 7%-7.2%. Yes, with my new hardware I have lots of CPU bandwidth available for other works. But I strongly believe a screen saver should not take 7% CPU usage with Intel i9500T Processor and 4 cores.

<img width="798" alt="MintResource" src="https://github.com/linuxmint/cinnamon/assets/19688360/357ffd22-8e2f-4285-aa0b-e7a7ddb5b288">

**Comment by snr-lab** [comment]
Hi @mtwebster 

My older system faced motherboard failure. I have purchased a new system with decent hardware configuration. The new system have capability to passthrough the GPU.

<img width="689" alt="SystemResource" src="https://github.com/linuxmint/cinnamon/assets/19688360/2de3ba9d-8b60-40ba-990a-d14656545729">

With the new hardware system performs much better. But the core issue remain same. When system is idle the cpu usage is 0.15% - 0.22%. But as soon as the screen saver is appears the cpu usage reaches to 7%-7.2%. Yes, with my new hardware I have lots of CPU bandwidth available for other works. But I strongly believe a screen saver should not take 7% CPU usage with Intel i9500T Processor and 4 cores.

<img width="637" alt="MintResource" src="https://github.com/linuxmint/cinnamon/assets/19688360/40b2b44e-56b2-4383-93db-dd235779c8c7">

**Comment by snr-lab** [comment]
Additionally I wanted to mention that when I am scaling the display to 125% the cpu usage is less than 1% as usual. But once the screen is locked the CPU usage reaches 80-90%.

**Comment by nima-1102** [comment]
+1

**Comment by snr-lab** [comment]
Hi @mtwebster

The problem is not about display scaling. It is understandable that display scaling consumes higher CPU.

Issue is screen saver consuming huge CPU irrespective of display scaling on or off. Using GPU things get better but problem remains same. As per me screen saver should be a very light weight program and it should consume tinny amount of resources.

I am eagerly waiting to hear from you.

**Comment by kylehase** [comment]
> My workaround is to disable lock screen in System settings -> Screen Saver

This dropped the CPU usage for my VMs instantly. I was expecting to have to restart or at least logout and log back in but the effect was immediate.

**Comment by ancientwizard** [comment]
I can't say its exactly related; but I too have had ugly CPU usage issues with cinnamon. With only a terminal CPU usage is okay, though it does spike with any small screen updates; I was unnoticed for a long time because I have 14 actual cores @>5Ghz. Launching pretty much any apps cause the cinnamon CPU usage to grow very fast. Together Firefox and VS-Code will make it sit at ~160% all the time. I switch to using the MATE manager/desktop. CPU issues solved.

**Comment by bersyl91** [comment]
> > My workaround is to disable lock screen in System settings -> Screen Saver
> 
> This dropped the CPU usage for my VMs instantly. I was expecting to have to restart or at least logout and log back in but the effect was immediate.

+1
CPU usage dropped from 96% to 1% instantly
I'm using Debian/Cinnamon in a Proxmox VM
